x-pack/plugins/ml/public/embeddables/anomaly_charts/anomaly_charts_setup_flyout.tsx

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { CoreStart } from 'kibana/public';
+import moment from 'moment';
+import { takeUntil } from 'rxjs/operators';
+import { from } from 'rxjs';
+import { i18n } from '@kbn/i18n';
+import { VIEW_BY_JOB_LABEL } from '../../application/explorer/explorer_constants';
+import {
+  KibanaContextProvider,
+  toMountPoint,
+} from '../../../../../../src/plugins/kibana_react/public';
+import { AnomalyChartsInitializer } from './anomaly_charts_initializer';
+import { JobSelectorFlyoutContent } from '../../application/components/job_selector/job_selector_flyout';
+import { AnomalyDetectorService } from '../../application/services/anomaly_detector_service';
+import { getInitialGroupsMap } from '../../application/components/job_selector/job_selector';
+import { getDefaultExplorerChartsPanelTitle } from './anomaly_charts_embeddable';
+import { getMlGlobalServices } from '../../application/app';
+import { HttpService } from '../../application/services/http_service';
+import { DashboardConstants } from '../../../../../../src/plugins/dashboard/public';
+import { AnomalyChartsEmbeddableInput } from '..';
+
+export async function resolveEmbeddableAnomalyChartsUserInput(
```

Comment:
It looks very similar to `resolveAnomalySwimlaneUserInput`. Please considered extracting some common logic (it's a significant part of this function).

Reply:
Updated here [`82e02c6` (#94396)](https://github.com/elastic/kibana/pull/94396/commits/82e02c681e1b54f0ad194e4cdc802e3aeac3f4e5) with changes from https://github.com/elastic/kibana/pull/95394